include/irr/asset/IAsset.h

```diff
@@ -141,9 +133,18 @@ class IAsset : virtual public core::IReferenceCounted
 
     IAsset() : isDummyObjectForCacheAliasing{false}, m_metadata{nullptr} {}
 
+	//! Returns whole size associated with an Asset and its data
+	/**
+		The size is used to determine compression level while writing process 
+		is performed. As you expect, the more size Asset has, the more compression
```

Comment:
Its actually necessary to reserve the correct amount of memory when writing to file, "conservative" here actually means that the value returned can be greater than memory actually needed.

What you've written about is actually a secondary functionality,
also, the bigger the asset the more likely it is to be compressed with a more expensive (slower) algorithm (replace the badly formed sentence ending).

Reply:
status?